api-controllers/ServiceFabrikApiController.js

```diff
@@ -427,16 +434,27 @@ class ServiceFabrikApiController extends FabrikBaseController {
           backup_guid: backupGuid,
           tenant_id: tenantId
         };
-        return this.backupStore
-          .getBackupFile(backupFileOptions);
+        return timeStamp ?
+          this.backupStore
+          .listBackupsOlderThan(backupFileOptions, new Date(timeStamp))
+          .then(oldBackupArray =>
+            _.sortBy(oldBackupArray, ['started_at']))
+          .then(sortedArray =>
+            _.findLast(sortedArray, backup => backup.state ===  CONST.OPERATION.SUCCEEDED))
+          .tap(successfulBackup => {
+            if (_.isEmpty(successfulBackup)) {
+              throw new NotFound(`No backup successful found for service instance '${instanceId}' before time_stamp ${new Date(timeStamp)} `);
```

Comment:
Maybe error msg can be: "Cannot restore service instance as no successful backup found for ..."